launch.php

```diff
@@ -46,7 +46,8 @@
     throw new moodle_exception('error:launchdirectaccess', 'applaunch');
 } else {
     // Attempt to launch the application.
-    header("Location: " . $appinstance->get_url());
+    $token = \mod_applaunch\external\ws_token::generate_user_key($cmid, $USER->id);
+    header("Location: " . $appinstance->get_url($token));
```

Comment:
given that this doesn't work the way we wanted I think we replace all this and template below with a simpler call to redirect()

Reply:
I've been debugging this. Moodle redirect doesn't seem to handle url's with different schema's

Reply:
ah yeah we did talk about this. all good